File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int>st;
        int cnt = 1,elem=0,ans=0;
        for(int i=0;i<nums.size();i++){
          st.insert(nums[i]);
        }
        for(int i : st){
            if(st.find(i-1) == st.end()){
               cnt = 1;
               elem = i;
               while(st.find(elem+1) != st.end()){
                elem = elem + 1;
                cnt++;
               }
               ans = max(ans,cnt);
            }
        }
        return ans;
    }
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        if(nums.length == 0) return 0;
        Arrays.sort(nums);
        int curr = nums[0], ans = 1;
        int cnt = 1;
        for(int i=1;i<nums.length;i++){

             if (nums[i] == curr) {
                continue;  // ignore duplicates
            }
           if((curr+1) == nums[i]){
            cnt++;
           }
           else if(curr+1 != nums[i]){
            
            cnt = 1;
           }
           curr = nums[i];
           ans = Math.max(ans,cnt);
        }
        return ans;
    }
}